### PR TITLE
Add terraform script for pubsub and update README.MD

### PR DESCRIPTION
**Pull request by weix0011**
Started a new pull request for practice.
Please review this PR instead of the old PR.

### CODE REVIEW

File: retail/retail-java-applications/README.MD [hunk 1]
```diff
@@ -201,3 +201,122 @@ VALUES
...
+
+```
+var.project_id
+  Project ID in GCP
+
+  Enter a value:
+```
+A list of resources planning to be created will be shown.
+
+```
```
Comment (tzehon) [inline retail/retail-java-applications/README.MD:262]: You can remove this portion

File: retail/retail-java-applications/README.MD [hunk 2]
```diff
@@ -201,3 +201,122 @@ VALUES
...
+ .
+ .
+ Plan: 8 to add, 0 to change, 0 to destroy.
+
+------------------------------------------------------------------------
+
+Note: You didn't specify an "-out" parameter to save this plan, so Terraform
+can't guarantee that exactly these actions will be performed if
+"terraform apply" is subsequently run.
+```
```
Comment (tzehon) [inline retail/retail-java-applications/README.MD:286]: You can remove this part

File: retail/retail-java-applications/README.MD [hunk 3]
```diff
@@ -201,3 +201,122 @@ VALUES
...
+- Pub/Sub
+
+### Install Terraform
+
+If you are using the Google Cloud Shell Terraform is already installed.
+
+Follow the instructions to [install the Terraform cli](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started).
+
+This repo has been tested on Terraform version `0.14.5` and the Google provider version `3.48.0`
+
```
Comment (tzehon) [inline retail/retail-java-applications/README.MD:218]: Add a section asking the user to look at the topic and subscription names in terraform.tfvars, explain that these are the defaults and that they can override them with their own names if desired

File: retail/retail-java-applications/variables.tf [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  type    = string
+}
+
+variable "transactions_inbound_sub" {
+  type    = string
+}
+
+variable "inventory_inbound_sub" {
+  type    = string
+}
```
Comment (tzehon) [inline retail/retail-java-applications/variables.tf:48]: Add a newline to satisfy the linter

File: retail/retail-java-applications/terraform.tfvars [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+
+topic_clickstream_inbound  = "Clickstream-inbound"
+topic_transactions_inbound = "Transactions-inbound"
+topic_inventory_inbound    = "Inventory-inbound"
+topic_inventory_outbound   = "Inventory-outbound"
+clickstream_inbound_sub    = "Clickstream-inbound-sub"
+transactions_inbound_sub   = "Transactions-inbound-sub"
+inventory_inbound_sub      = "Inventory-inbound-sub"
```
Comment (tzehon) [inline retail/retail-java-applications/terraform.tfvars:24]: Add a newline to satisfy the linter

**Comment by tzehon** [comment]
Looking good, some minor comments.

File: retail/retail-java-applications/README.MD (hunk 2 above)
Comment (weix0011) [reply retail/retail-java-applications/README.MD]: Removed accordingly.

File: retail/retail-java-applications/README.MD (hunk 3 above)
Comment (weix0011) [reply retail/retail-java-applications/README.MD:218]: Added a section about the resource names. Please take a look at it. Thank you.

File: retail/retail-java-applications/variables.tf (hunk 4 above)
Comment (weix0011) [reply retail/retail-java-applications/variables.tf]: Added a new line

File: retail/retail-java-applications/terraform.tfvars (hunk 5 above)
Comment (weix0011) [reply retail/retail-java-applications/terraform.tfvars]: Added a new line

File: .gitignore [hunk 6]
```diff
@@ -90,3 +90,8 @@ sdks/python/postcommit_requirements.txt
 
 .pytest_cache
 .pytest_cache/**/*
+
+# Terraform intermediate artifacts
+.terraform/
+.terraform.lock.hcl
+*.tfstate
```
Comment (tzehon) [inline .gitignore:97]: The current .gitignore doesn't ignore terraform.tfstate.backup files, please modify it. Now that you're more familiar with .gitignore and Terraform in general, you can look at https://www.toptal.com/developers/gitignore which is a useful site to help generate .gitignore files. 

I wouldn't suggest always taking 100% of their generated .gitignore however, examine each line individually and take those that make sense in the context of each commit/project.

**Review by tzehon — CHANGES_REQUESTED** [review]
One comment on .gitignore, everything else looks good

File: .gitignore (hunk 6 above)
Comment (weix0011) [reply .gitignore:97]: Added *.tfstate.* to ignore terraform.tfstate.backup files. After looking at the generated .gitigore file, no additional file is included according to the following reasons.
1. crash.log - no crash log is expected as this project is straight forward
2. override.tf - there is no need to override existing tf files for now
3. tfplan files - tfplan files are generated according to users' need and could be named differently
Thank you for reviewing! :)

**Review by tzehon — APPROVED** [review]
LGTM

**Comment by tzehon** [comment]
Click the big green 'Merge pull request' button, use the default option, and then leave it for now. I'll walk you through some extra setup you need to do for your GitHub account tomorrow so you can raise a PR for merging to upstream

**Comment by weix0011** [comment]
First merged PR! Thank you Tze Hon for your helpful reviews!

**Pull request merged**